### ISSUE TITLE
Improved functionality

### ISSUE THREAD

**Issue by deviouspk** [rdohms/phpunit-arraysubset-asserts#6]
Could you take a look at the explanation for the frustrations with the deprecated method i posted:
https://github.com/sebastianbergmann/phpunit/issues/3494#issuecomment-479905197

Would this be something you are interested in to implement?

**Comment by rdohms** [comment]
@deviouspk yes, I had originally planned to review and improve for here. Ideally without breaking BC and by introducing new asserts. I saw your description and it looks like a good starting point to discuss.

**Comment by jagarsoft** [comment]
Hello!!

How about you use an array of options? like this:

[
'checkForObjectIdentity' => true,
'strict' => false,
'order' => true
]

**Comment by JakeQZ** [comment]
👍 for providing this extension.

For the case where both arrays are non-associative, could the `Constraint` simply check for this, and if both are indexed (or empty) arrays, ignore the keys?  This would avoid any need for an extra parameter.
